pkg/query-service/app/clickhouseReader/reader.go

```diff
@@ -3152,6 +3152,22 @@ func (r *ClickHouseReader) GetTotalSpans(ctx context.Context) (uint64, error) {
 	return totalSpans, nil
 }
 
+func (r *ClickHouseReader) GetUserRole(ctx context.Context, groupId string) (string, error) {
+	role, err := dao.DB().GetGroup(context.Background(), groupId)
+	if err != nil || role == nil {
+		return "", err
+	}
+	return role.Name, nil
+}
+
+func (r *ClickHouseReader) GetUserCount(ctx context.Context) (int, error) {
+	users, err := dao.DB().GetUsers(ctx)
+	if err != nil {
+		return 0, err
+	}
+	return len(users), nil
+}
```

Comment:
Let's not add these to the ClickHouseReader interface. Please use dao in the telemetry instance to access the info. We are in the process of removing non-ClickHouse things from the ClickHouse reader interface.

Reply:
There will be cyclic dependency if we try to add dao

Reply:
We should fix the cyclic dependency.

Reply:
Yeah, that would involve creating an interface for dao, like we have for reader.
It would be a significant change, I think we are already working on improving query service structure in parallel so let's not block this PR.

Reply:
since your bandwidth is prioritized elsewhere I took a pull and moved them to callbacks. this will get count and role details without adding new functions to the CH reader interface.